Tools/autotest/arducopter.py

```diff
@@ -924,7 +924,33 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
     wait_mode(mav, 'LOITER')
     return ret
 
-
+def fly_motor_fail(mavproxy, mav, fail_servo=0, fail_mul=1.0, holdtime=30):
+    """Fly a mission from a file."""
+    global homeloc
+    global num_wp
+    if (fail_servo < 0) or (fail_servo > 6):
+        fail_servo = 0
+    mavproxy.send('switch 5\n')  # loiter mode
+    wait_mode(mav, 'LOITER')
+    print("test: Fly a mission from 1 to {0}, with motor {1} efficiency at {2}%".format(num_wp,fail_servo+1,fail_mul))
+    change_alt(mavproxy,mav,alt_min=50)
+    mavproxy.send('param set SIM_ENGINE_FAIL %u\n' % fail_servo)
+    mavproxy.send('param set SIM_ENGINE_MUL %f\n' % fail_mul)
+    tstart = get_sim_time(mav)
+    while get_sim_time(mav) < tstart + holdtime:
+        data = mav.recv_match(type='SERVO_OUTPUT_RAW', blocking=True)
+        servo_list=[data.servo1_raw, data.servo2_raw, data.servo3_raw, data.servo4_raw, data.servo5_raw, data.servo6_raw, data.servo7_raw, data.servo8_raw]
+        output_message = "PWM output per motor: "
+        for i, val in enumerate(servo_list):
+            if val>0: #Prevent printing of unused servo channels
+                if i==fail_servo:
+                    output_message+="(failed) "
+                output_message+= "servo " + str(i+1) + " " + str(val) + "; "
+        print(output_message)
+    #Reset motor efficiency
```

Comment:
In this case SIM_ENGINE_FAIL only defines the servo to which the efficiency multiplier SIM_ENGINE_MUL is applied to, so un-failing the motor comes down to set the multiplier to one, but I can also reset SIM_ENGINE_FAIL to a given servo (0 for example)